internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -21,8 +23,110 @@ var _ = Describe("NGINX Runtime Manager", func() {
 		mgr = NewManagerImpl(&ngxclient.NginxClient{}, nil, logr.New(GinkgoLogr.GetSink()))
 		Expect(mgr.IsPlus()).To(BeTrue())
 	})
+
+	Describe("ManagerImpl", Ordered, func() {
+		var (
+			mockedManager Manager
+			serverMocks   []ngxclient.UpstreamServer
+			serversMock   ngxclient.UpstreamServer
+
+			clientMock *ngxclient.NginxClient
+		)
+
+		var (
+			MaxConnsMock = 150
+			MaxFailsMock = 20
+			WeightMock   = 10
+			BackupMock   = false
+			DownMock     = false
+		)
+
+		BeforeAll(func() {
+
+			serversMock = ngxclient.UpstreamServer{
+				ID:          1,
+				Server:      "unknown",
+				MaxConns:    &MaxConnsMock,
+				MaxFails:    &MaxFailsMock,
+				FailTimeout: "test",
+				SlowStart:   "test",
+				Route:       "test",
+				Backup:      &BackupMock,
+				Down:        &DownMock,
+				Drain:       false,
+				Weight:      &WeightMock,
+				Service:     "",
+			}
+
+			serverMocks = []ngxclient.UpstreamServer{
+				serversMock,
+			}
+
+			httpClient := &http.Client{
+				Transport:     http.DefaultTransport,
+				CheckRedirect: http.DefaultClient.CheckRedirect,
+				Jar:           http.DefaultClient.Jar,
+				Timeout:       time.Second * 4,
+			}
+
+			clientMock, _ = ngxclient.NewNginxClient("test", client.WithHTTPClient(httpClient))
```

Comment:
Per my other comment, this isn't technically a mocked client. This is still initializing a real client, which is looking for the endpoint "test", which doesn't exist.

We'll need an ngxclient.FakeClient() passed to the manager that implements the same methods that the real client implements. This means interfaces and the fake client implementation will have to be added to that library.

Reply:
@sjberman thanks for amazingly fast feedback, 

so u meant to use counterfeiter for this one ?  
https://github.com/nginxinc/nginx-plus-go-client/blob/main/client/nginx.go 

and then use that one from "nginx-plus-go-client" in our "nginx-gateway-fabric" 
so to generate mocks for NginxClient struct, we actually first need to create an interface that includes methods for our client ?  

thanks !

Reply:
Correct, we'll first have to define an interface that contains the methods that the [client](https://github.com/nginxinc/nginx-plus-go-client/blob/main/client/nginx.go) has defined, then use counterfeiter on that interface to generate the mocks. And I misspoke in my earlier comments, we actually shouldn't need to define that interface in that repository, we can define it in our own `nginx-gateway-fabric` repository. Then we would just have to update the Manager to accept that interface, instead of the literal client, so that way in the unit tests we can supply the fake client that also implements the interface.

Reply:
sure, that make sense, 
tnx for providing a lot of valuable informations, 
Im gonna proceed accordingly and add proper updates